### PR TITLE
Fix "whitespace" words - docs/csharp

### PR DESCRIPTION
**Pull request by lucasnobredev**
## Fix "whitespace" words - docs/csharp

This PR came out from this [issue](https://github.com/dotnet/docs/issues/4439). It's the second PR and it's all about docs/csharp.

## Brief explanation
We should use **white-space** for adjectives and **white space** for nouns.

## Suggested Reviewers
@mairaw @rpetrusha @wstaelens

### CODE REVIEW

**Review by BillWagner — APPROVED** [review]
Thanks @lucasnobredev I gave it a careful 👓 

It looks right to me. I'm going to wait on Ron's review before merging, because I often get this wrong too. 😥

File: docs/csharp/pattern-matching.md
Comment (rpetrusha) [inline docs/csharp/pattern-matching.md]: In the code file, a comment reads "// whitespace". Could you change it to "// white space"?

File: docs/csharp/programming-guide/concepts/linq/preserving-white-space-while-serializing.md
Comment (rpetrusha) [inline docs/csharp/programming-guide/concepts/linq/preserving-white-space-while-serializing.md]: on line 19, "white space text nodes" --> "white-space text nodes"

File: docs/csharp/programming-guide/concepts/linq/programming-with-nodes.md
Comment (rpetrusha) [inline docs/csharp/programming-guide/concepts/linq/programming-with-nodes.md]: there are two occurrences of "white space text nodes" --> "white-space text nodes"

File: docs/csharp/properties.md
Comment (rpetrusha) [inline docs/csharp/properties.md]: Unrelated to your PR: Could you delete the comma between "blank" and "or" both here and in line 149?

**Review by rpetrusha — CHANGES_REQUESTED** [review]
Thank you so much for continuing to work on correcting the occurrences of "whitespace" in the documentation, @lucasnobredev. We really appreciate it. There's one issue unrelated two your PR that I'm hoping you can fix, as well as a couple of additional white-space-related changes.

File: docs/csharp/properties.md
Comment (lucasnobredev) [reply docs/csharp/properties.md]: Done!

File: docs/csharp/programming-guide/concepts/linq/preserving-white-space-while-serializing.md
Comment (lucasnobredev) [reply docs/csharp/programming-guide/concepts/linq/preserving-white-space-while-serializing.md]: Done!

File: docs/csharp/pattern-matching.md
Comment (lucasnobredev) [reply docs/csharp/pattern-matching.md]: Done!

**Review by lucasnobredev — COMMENTED** [review]
It's done!

**Review by rpetrusha — APPROVED** [review]

**Comment by rpetrusha** [comment]
Thanks for making the additional changes, @lucasnobredev, as well as for doing such excellent work overall. I've approved your PR, but our repo is undergoing some changes at the moment (see issue #4789). We’ll merge your PR as soon as the changes are completed.

**Pull request merged**
